pandas/_libs/tslibs/c_timestamp.pxd

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+
+from cpython.datetime cimport datetime
+
+from numpy cimport int64_t
+
+cdef class _Timestamp(datetime):
+    cdef readonly:
+        int64_t value, nanosecond
+        object freq
+        list _date_attributes
+    cpdef bint _get_start_end_field(self, str field)
```

Comment:
future PR we should rename the 'private' functions to public if they are actually exposed (the _ leading ones)